priv/repo/migrations/20220418150726_add_publisher_to_projects_and_products.exs

```diff
@@ -0,0 +1,46 @@
+defmodule Oli.Repo.Migrations.AddPublisherToProjectsAndProducts do
+  use Ecto.Migration
+
+  import Ecto.Query, warn: false
+
+  alias Oli.Delivery.Sections.Section
+  alias Oli.Inventories
+  alias Oli.Inventories.Publisher
+  alias Oli.Repo
+
+  require Logger
+
+  def up do
+    default_publisher_attrs = %{name: Inventories.default_publisher_name(), email: "publisher@cmu.edu"}
+
+    case Inventories.find_or_create_publisher(default_publisher_attrs) do
+      {:ok, %Publisher{id: publisher_id}} ->
+        alter table(:sections) do
+          add :publisher_id, references(:publishers)
```

Comment:
should this be null false + defaulted to this publisher as well?

Reply:
We can't add a `null: false` constraint here since we don't want publisher to be required in all sections. We only want that for products (`type == :blueprint`)